docs/docs/propeller/streams-sink.md

```diff
@@ -0,0 +1,37 @@
+# StreamsSink Architecture
+
+## AsyncQueue
+
+The StreamsSink uses an AsyncQueue that provides a node-friendly way to handle a
+queue of items without blocking waits.
+
+It is backed by a basic FIFO Queue, implemented using a linked list. When an
+item is added to the AsyncQueue, it first checks if there are any pending
+retrieval requests. If there are, it fulfills the oldest request. If not, it
+queues the item. The dequeueing operation retrieves an item from the queue. If
+the queue is empty, it waits asynchronously until an item is available or until
+an abort signal is triggered, which would stop the wait and reject the promise
+
+## Receiving batches
+
+For every event within a batch, the StreamsSink checks if its associated Stream
+is already in the processing queue. If it is, the event is combined with it,
+ensuring it's processed alongside previously queued events.
+
+The set of streams associated with the batch are stored in a `Set` so we can
```

Comment:
set used twice
Per batch, we establish a Set of the streams for which there were events. Whenever a handler completes processing work for a stream, it gets ticked off; the last one results in flagging the batch complete, which the FeedSource then uses to a) update the checkpoint to record the progress attained b) trigger commencement of reading of a replacement batch per the max read ahead policy